source/core/shape.cc

```diff
@@ -0,0 +1,427 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 3.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+#include <core/shape.h>
+
+template <int dim>
+double
+Shape<dim>::displaced_volume(const double fluid_density)
+{
+  StandardExceptions::ExcNotImplemented();
+  return 1.;
+}
+
+template <int dim>
+Point<dim>
+Shape<dim>::align_and_center(const Point<dim> &evaluation_point) const
+{
+  // Translation and rotation to standard position and orientation for
+  // distance calculations
+  Point<dim> center_of_rotation = position;
+
+  Point<dim> rotated_point;
+  Point<dim> translated_point;
+
+  // Rotation from the solid orientation
+  // Angular position around x, y and z axis
+  Tensor<1, 3> theta = orientation;
+  Point<dim>   centralized_point;
+  centralized_point              = evaluation_point - center_of_rotation;
```

Comment:
Is there any specific reason to have centralized_point and centralized_rotated here?  Apparently you use only the centralized_rotated here.

Reply:
Yes, the centralized point is the one to be rotated, and it is updated after each rotation around one axis. The centralized rotated point is the result of each rotation, and it is initialized in case no rotation is performed. I added this as a comment, I agree that it wasn't clear :)